### PR TITLE
CoAP examples: make example a host instead of router

### PR DESCRIPTION
**Pull request by miri64**
There was some confusion that there was no host example. Since the microcoap example also contains RPL to be able to route which makes the example more complicated (and additionally isn't even documented in its README), I thought it might be a good idea to remove routing capabilities from that example.

### CODE REVIEW

**Comment by miri64** [comment]
```
text    data    bss     dec     BOARD/BINDIRBASE

-2832   0       -1792   -4624   arduino-due
35940   564     16356   52860   old
33108   564     14564   48236   new

-5186   -128    -1493   -6807   arduino-duemilanove
45414   4346    12188   61948   old
40228   4218    10695   55141   new

-5190   -126    -1493   -6809   arduino-mega2560
46566   4362    12200   63128   old
41376   4236    10707   56319   new

-5186   -128    -1493   -6807   arduino-uno
45414   4338    12188   61940   old
40228   4210    10695   55133   new

-2944   0       -1792   -4736   arduino-zero
38000   500     16420   54920   old
35056   500     14628   50184   new

-4616   0       4       -4612   avsextrem
55744   500     97797   154041  old
51128   500     97801   149429  new

-4608   0       -1920   -6528   cc2538dk
49792   500     19360   69652   old
45184   500     17440   63124   new

-2832   0       -1796   -4628   cc2650stk
35548   500     16540   52588   old
32716   500     14744   47960   new

-2832   0       -1792   -4624   ek-lm4f120xl
37196   500     16284   53980   old
34364   500     14492   49356   new

-2832   0       -1792   -4624   f4vi1
38068   504     16412   54984   old
35236   504     14620   50360   new

-4612   0       -1924   -6536   fox
50624   500     19492   70616   old
46012   500     17568   64080   new

-2832   0       -1792   -4624   frdm-k64f
36860   1528    16844   55232   old
34028   1528    15052   50608   new

-4612   0       -1924   -6536   iotlab-a8-m3
50652   500     19492   70644   old
46040   500     17568   64108   new

-2832   0       -1796   -4628   iotlab-m3
36088   500     16420   53008   old
33256   500     14624   48380   new

-2832   0       -1792   -4624   limifrog-v1
37216   500     16404   54120   old
34384   500     14612   49496   new

-2832   0       -1792   -4624   mbed_lpc1768
35720   500     16276   52496   old
32888   500     14484   47872   new

-7608   0       4       -7604   msba2
76028   500     97797   174325  old
68420   500     97801   166721  new

-2832   0       -1792   -4624   msbiot
38420   504     16444   55368   old
35588   504     14652   50744   new

-4612   0       -1920   -6532   mulle
54220   1564    19992   75776   old
49608   1564    18072   69244   new

-8464   0       -1716   -10180  native
111052  460     85588   197100  old
102588  460     83872   186920  new

-3296   0       -1792   -5088   nrf52dk
46968   500     34260   81728   old
43672   500     32468   76640   new

-2944   0       -1792   -4736   nucleo-f091
37424   500     16404   54328   old
34480   500     14612   49592   new

-2832   0       -1796   -4628   nucleo-f103
36048   500     16428   52976   old
33216   500     14632   48348   new

-2832   0       -1792   -4624   nucleo-f207
38556   504     16460   55520   old
35724   504     14668   50896   new

-2832   0       -1792   -4624   nucleo-f303
36448   500     16412   53360   old
33616   500     14620   48736   new

-1552   0       -316    -1868   nucleo-f334
34576   504     9748    44828   old
33024   504     9432    42960   new

-2832   0       -1792   -4624   nucleo-f401
38040   504     16412   54956   old
35208   504     14620   50332   new

-2832   0       -1792   -4624   nucleo-f446
38040   504     16412   54956   old
35208   504     14620   50332   new

-2832   0       -1792   -4624   nucleo-l1
37096   500     16396   53992   old
34264   500     14604   49368   new

-4608   0       -1920   -6528   openmote-cc2538
50416   500     19616   70532   old
45808   500     17696   64004   new

-4612   0       -1920   -6532   pba-d-01-kw2x
52308   1528    20020   73856   old
47696   1528    18100   67324   new

-4616   0       4       -4612   pttu
55948   500     97797   154245  old
51332   500     97801   149633  new

-4608   0       -1920   -6528   remote-pa
50984   500     19560   71044   old
46376   500     17640   64516   new

-4608   0       -1920   -6528   remote-reva
51016   500     19600   71116   old
46408   500     17680   64588   new

-4608   0       -1920   -6528   remote-revb
51024   500     19600   71124   old
46416   500     17680   64596   new

-2944   0       -1792   -4736   saml21-xpro
37436   500     16396   54332   old
34492   500     14604   49596   new

-4772   0       -1924   -6696   samr21-xpro
53352   500     19496   73348   old
48580   500     17572   66652   new

-2832   0       -1792   -4624   seeeduino_arch-pro
35724   500     16276   52500   old
32892   500     14484   47876   new

-2832   0       -1796   -4628   slwstk6220a
35740   500     16404   52644   old
32908   500     14608   48016   new

-2944   0       -1792   -4736   sodaq-autonomo
38052   500     16436   54988   old
35108   500     14644   50252   new

-2832   0       -1792   -4624   stm32f3discovery
36536   500     16412   53448   old
33704   500     14620   48824   new

-2832   0       -1792   -4624   stm32f4discovery
38344   504     16428   55276   old
35512   504     14636   50652   new

-2832   0       -1792   -4624   udoo
35936   564     16356   52856   old
33104   564     14564   48232   new

-5186   -126    -1493   -6805   waspmote-pro
46186   4354    12192   62732   old
41000   4228    10699   55927   new
```

Maybe I can even remove some boards from `BOARD_INSUFFIENT_MEMORY` now :-).

**Comment by miri64** [comment]
Yes, and done.

**Comment by miri64** [comment]
Also adapted nanocoap example.

**Comment by miri64** [comment]
#6030 required.

**Comment by kaspar030** [comment]
please rebase

**Comment by kaspar030** [comment]
I think this makes sense, as routing and RPL still has to be manually configured, right?

**Comment by miri64** [comment]
> I think this makes sense, as routing and RPL still has to be manually configured, right?

Can you clarify how you mean that? A host has no routing capability so there is no routing nor RPL. So you see how I am a little confused? ;-)

**Comment by kaspar030** [comment]
> Can you clarify how you mean that? A host has no routing capability so there is no routing nor RPL. So you see how I am a little confused? ;-)

I mean, even if routing & RPL is compiled in (like currently in the examples), it doesn't do much until configured, right? And as the coap examples don't include a shell, there's no way of configuring anything.

**Comment by miri64** [comment]
Yes (though there is a possibility for auto-init). The behavior regarding NDP is also different for hosts compared to routers (routers advertise themselves, hosts search for routers)

**Comment by kaspar030** [comment]
> Yes (though there is a possibility for auto-init).

My point being, all this time that functionality has been compiled into the CoAP examples without an actual way of using it, so this PR definately makes sense.

**Comment by miri64** [comment]
Rebased

File: examples/microcoap_server/Makefile [hunk 1]
```diff
@@ -7,21 +7,16 @@ BOARD ?= native
 # This has to be the absolute path to the RIOT base directory:
 RIOTBASE ?= $(CURDIR)/../..
 
-BOARD_INSUFFICIENT_MEMORY := airfy-beacon chronos msb-430 msb-430h nrf51dongle \
-                          nrf6310 pca10000 pca10005 spark-core \
-                          stm32f0discovery telosb weio wsn430-v1_3b wsn430-v1_4 \
-                          yunjia-nrf51822 z1 nucleo-f072 nucleo-f030 nucleo-f070 \
-                          microbit
```
Comment (miri64) [inline examples/microcoap_server/Makefile]: Yes, I tested `microbit` ;-)

**Comment by miri64** [comment]
Oops, how isn't this merged yet? Rebased.

**Comment by OlegHahm** [comment]
Please squash

**Comment by miri64** [comment]
Rebased and added `nucleo-f042` to list of boards with insufficient memory, as reported by murdock.

**Comment by PeterKietzmann** [comment]
Too many green lights not to push this button

**Pull request merged**